trackparser.py

```diff
@@ -97,41 +147,38 @@ def parseCurve(attrs):
 
 
 	if attrs['type'] == 'lft':
-		center = (
-			startPoint[0] - lastDirection[1] * radius,
-			startPoint[1] + lastDirection[0] * radius
+		center = Point(
+			startPoint[0] - lastDirection[1] * radiusX,
+			startPoint[1] + lastDirection[0] * radiusY
 		)
-		# rotate left
-		directionDegree = (directionDegree + arc) % 360
 	elif attrs['type'] == 'rgt':
-		center = (
-			startPoint[0] + lastDirection[1] * radius,
-			startPoint[1] - lastDirection[0] * radius
+		center = Point(
+			startPoint[0] + lastDirection[1] * radiusX,
+			startPoint[1] - lastDirection[0] * radiusY
 		)
-		# rotate right
-		directionDegree = (directionDegree - arc) % 360
 	else:
 		logger.warn('Wrong direction')
 
-	logger.debug('Curve: center %s, radius %s, arc %s' % (center, radius, arc))
+
+	logger.debug('Curve: center %s, radius %s, %s, arc %s' % (center, radiusX, radiusY, arc))
 	helpNodes.append(center)
 
-	curDegree = degreeStepSize
-	while curDegree < arc:
-		curRad = math.radians(curDegree)
-		newNode = (
-			center[0] + ( startPoint[0] - center[0] ) * np.cos(rotationDirection * curRad) + ( center[1] - startPoint[1] ) * np.sin(rotationDirection * curRad),
-			center[1] + ( startPoint[1] - center[1] ) * np.cos(rotationDirection * curRad) + ( startPoint[0] - center[0] ) * np.sin(rotationDirection * curRad)
-		)
-		nodes.append(newNode)
-		curDegree += degreeStepSize
 
-	# add end node
-	newNode = (
-		center[0] + ( startPoint[0] - center[0] ) * np.cos(rotationDirection * radArc) + ( center[1] - startPoint[1] ) * np.sin(rotationDirection * radArc),
-		center[1] + ( startPoint[1] - center[1] ) * np.cos(rotationDirection * radArc) + ( startPoint[0] - center[0] ) * np.sin(rotationDirection * radArc)
-	)
-	nodes.append(newNode)
+	curveNodes = sampleEllipse(center, radiusX, radiusY, arc, degreeStepSize)
+
+	if attrs['type'] == 'rgt':
+		curveNodes = [ flipHorizontally(node, center.x) for node in curveNodes ]
+
+	curveNodes = [ rotatePoint(center, node, directionDegree-90) for node in curveNodes ]
```

Comment:
To attach the ellipse to the end of the previous segment we have to rotate it into place.